rebase_automation.sh

```
@@ -0,0 +1,527 @@
+    # Function to safely replace versions (avoiding URLs and specific patterns)
+    safe_replace_version() {
+        local file="$1"
+        local old_version="$2"
+        local new_version="$3"
+        local context="$4"
+        
+        # Skip if file doesn't exist or isn't readable
+        [[ ! -f "$file" || ! -r "$file" ]] && return
+        
+        # Create a temporary file for processing
+        local temp_file=$(mktemp)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)